src/hooks/useSubaccount.tsx

```diff
@@ -909,6 +911,69 @@ const useSubaccountContext = ({ localDydxWallet }: { localDydxWallet?: LocalWall
     [localDydxWallet, compositeClient]
   );
 
+  const registerAffiliate = useCallback(
+    async (affiliate: string) => {
+      if (!compositeClient) {
+        throw new Error('client not initialized');
+      }
+      if (!subaccountClient?.wallet?.address) {
+        throw new Error('wallet not initialized');
+      }
+      if (affiliate === subaccountClient?.wallet?.address) {
+        throw new Error('affiliate can not be the same as referree');
+      }
+      try {
+        const response = await compositeClient?.validatorClient.post.registerAffiliate(
+          subaccountClient,
+          affiliate
+        );
+        return response;
+      } catch (error) {
+        log('useSubaccount/registerAffiliate', error);
+        return undefined;
+      }
+    },
+    [subaccountClient, compositeClient]
+  );
+
+  const latestReferrer = useAppSelector(getLatestReferrer);
+  const { data: referredBy, isFetched: isReferredByFetched } = useReferredBy();
+
+  useEffect(() => {
+    if (!subaccountClient) return;
+
+    if (dydxAddress === latestReferrer) {
+      dispatch(removeLatestReferrer());
+      return;
+    }
+    if (
+      latestReferrer &&
+      dydxAddress &&
+      usdcCoinBalance &&
+      parseFloat(usdcCoinBalance.amount) > AMOUNT_USDC_BEFORE_REBALANCE &&
+      isReferredByFetched &&
+      !referredBy
+    ) {
+      registerAffiliate(latestReferrer);
+      dispatch(removeLatestReferrer());
```

Comment:
it might also make sense to dispatch this action inside the mutation too, since if there is an error with the network call above, we might not want to remove it from localStorage just yet. either that or add some retry logic to useMutation xD but.. this is not a big deal